src/DataStructures/Mesh.hpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Mesh.
+
+#pragma once
+
+#include "DataStructures/Index.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+template <size_t Dim>
+class Mesh : public Index<Dim> {
+ public:
+  explicit Mesh(
+      const size_t uniform_extents = std::numeric_limits<size_t>::max(),
+      Spectral::Basis uniform_basis = Spectral::Basis::Legendre,
+      Spectral::Quadrature uniform_quadrature =
+          Spectral::Quadrature::GaussLobatto)
+      : Index<Dim>(std::move(uniform_extents)) {
+    bases_.fill(uniform_basis);
+    quadratures_.fill(uniform_quadrature);
+  }
+
+  explicit Mesh(std::array<size_t, Dim> extents,
+                Spectral::Basis uniform_basis = Spectral::Basis::Legendre,
+                Spectral::Quadrature uniform_quadrature =
+                    Spectral::Quadrature::GaussLobatto)
+      : Index<Dim>(std::move(extents)) {
+    bases_.fill(uniform_basis);
+    quadratures_.fill(uniform_quadrature);
+  }
+
+  explicit Mesh(std::array<size_t, Dim> extents,
```

Comment:
no explicit

Reply:
could you explain a bit the reasoning where we want this and where we don't?

Reply:
http://en.cppreference.com/w/cpp/language/explicit

we don't want converting constructors in general